src/meshes/p4est_mesh.jl

```diff
@@ -1021,9 +1136,267 @@ function cubed_sphere_mapping(xi, eta, zeta, inner_radius, thickness, direction)
+    curved_check[4] = parse(Int, current_line[5])
+    if sum(curved_check) == 0
+      # Create the node coordinates on this particular element
+      calc_node_coordinates!(node_coordinates, tree, nodes, cornerNodeVals)
+    else
+      # Quadrilateral element has at least one curved side
+      # Flip node ordering to make sure the element is right-handed for the interpolations
+      m1 = 1
+      m2 = 2
+      @views tempNodes[1, :] .= cornerNodeVals[4, :]
+      @views tempNodes[2, :] .= cornerNodeVals[2, :]
+      @views tempNodes[3, :] .= cornerNodeVals[3, :]
+      @views tempNodes[4, :] .= cornerNodeVals[1, :]
+      for i in 1:4
+        if curved_check[i] == 0
+          # When curved_check[i] is 0 then the "curve" from cornerNode(i) to cornerNode(i+1) is a
+          # straight line. So we evaluate a linear interpolant between the two points at each of the nodes.
+          for k in 1:nnodes
+            curve_vals[k, 1] = linear_interpolate(nodes[k], tempNodes[m1, 1], tempNodes[m2, 1])
+            curve_vals[k, 2] = linear_interpolate(nodes[k], tempNodes[m1, 2], tempNodes[m2, 2])
+          end
+        else
+          # When curved_check[i] is 1 this curved boundary information is supplied by the mesh
+          # generator. So we just read it into a work array
+          for k in 1:nnodes
+            file_idx += 1
+            current_line = split(file_lines[file_idx])
+            curve_vals[k, 1] = parse(RealT,current_line[2])
+            curve_vals[k, 2] = parse(RealT,current_line[3])
+          end
+        end
+        # Construct the curve interpolant for the current side
+        surface_curves[i] = CurvedSurfaceT(nodes, bary_weights, copy(curve_vals))
+        # Indexing update that contains a "flip" to ensure correct element orientation.
+        # If we need to construct the straight line "curves" when curved_check[i] == 0
+        m1 += 1
+        if i == 3
+          m2 = 1
+        else
+          m2 += 1
+        end
+      end
+      # Create the node coordinates on this particular element
+      calc_node_coordinates!(node_coordinates, tree, nodes, surface_curves)
+    end
+    # move file index to the next tree
+    file_idx += 1
+   end
+
+  return file_idx
+end
+
+
+# Calculate physical coordinates of each element of an unstructured mesh read
+# in from a HOHQMesh file. This calculation is done with the transfinite interpolation
+# routines found in `transfinite_mappings_3d.jl`
+function calc_tree_node_coordinates!(node_coordinates::AbstractArray{<:Any, 5},
+                                     file_lines::Vector{String}, nodes, vertices, RealT)
+  # Get the number of trees and the number of interpolation nodes
+  n_trees = last(size(node_coordinates))
+  nnodes = length(nodes)
+
+  # Setup the starting file index to read in element indices and the additional
+  # curved boundary information provided by HOHQMesh.
+  file_idx = findfirst(occursin.("mesh polynomial degree", file_lines)) + 1
```

Comment:
Same optimization with `contains` as above?

Reply:
See above, it gives an error